src/libs/antares/study/load-params.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include "study.h"
+
+using namespace Yuni;
+
+
+namespace Antares
+{
+namespace Data
+{
+
+bool Study::initializeInternalData(const StudyLoadOptions& options)
```

Comment:
Apparently, method **initializeInternalData** does 3 (at least) things : 
- it initializes **parameters** (it should be done by a method of class **Parameters**, or better : in the **Parameters** constructor, if possible)
- it initializes the calendar (same remark as above, with class **Date::Calendar**)
- it makes a check (should be moved with other checks : to be explored)

Each bullet should be separated from the others in a function / method
This split would have few consequences **test-study-load-internal.cpp**.